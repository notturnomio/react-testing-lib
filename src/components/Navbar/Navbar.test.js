/* eslint-disable testing-library/no-unnecessary-act */
import { screen } from '@testing-library/react';
import { renderWithRouter } from '../../tests/helpers/renderWithRouter';
import Navbar from './Navbar';
import { act } from 'react-dom/test-utils';
import userEvent from '@testing-library/user-event';

describe('TEST Navbar', () => {
  test('tests main link', async () => {
    await act(async () => {
      renderWithRouter(<Navbar />);
    });
    const mainLink = screen.getByTestId('main-link');
    await act(async () => {
      userEvent.click(mainLink);
    });
    expect(await screen.findByTestId('main-page')).toBeInTheDocument();
  });
  test('tests about link', async () => {
    await act(async () => {
      renderWithRouter(<Navbar />);
    });
    const aboutLink = screen.getByTestId('about-link');
    await act(async () => {
      userEvent.click(aboutLink);
    });
    expect(await screen.findByTestId('about-page')).toBeInTheDocument();
  });
  test('tests users link', async () => {
    await act(async () => {
      renderWithRouter(<Navbar />);
    });
    const usersLink = screen.getByTestId('users-link');
    await act(async () => {
      userEvent.click(usersLink);
    });
    expect(await screen.findByTestId('users-page')).toBeInTheDocument();
  });
});
