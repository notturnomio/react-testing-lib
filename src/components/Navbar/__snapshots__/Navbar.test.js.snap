// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST Navbar checks sample page elements 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;

exports[`TEST Navbar renders links 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;

exports[`TEST Navbar renders nav links 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;
